### PR TITLE
added additional signInWithPopup method to fix signInWithProvider problem in web app

### PR DESCRIPTION
**Pull request by raymon-io**
This package works well in mobile app but web app produces error "signInWithProvider() is not implemented". 

I believe this is expected as official documents suggests to use signInWithPopup for other oauth sign in methods such as for Microsoft and Twitter. https://github.com/firebase/flutterfire/blob/master/docs/auth/federated-auth.md#microsoft

### CODE REVIEW

File: lib/src/provider.dart [hunk 1]
```diff
@@ -48,7 +48,11 @@ class OidcProvider extends OAuthProvider {
           ),
         );
       } else {
-        credential = await auth.signInWithProvider(firebaseAuthProvider);
+        if (kIsWeb) {
```
Comment (Rexios80) [inline lib/src/provider.dart]: Please un-nest this into an `else if`/`else` on the enclosing `if` statement

**Review by Rexios80 — CHANGES_REQUESTED** [review]

File: lib/src/provider.dart (hunk 1 above)
Comment (raymon-io) [reply lib/src/provider.dart]: Done

**Review by Rexios80 — APPROVED** [review]

**Comment by Rexios80** [comment]
Released in `0.2.1`

**Comment by Rexios80** [comment]
Thanks for the contribution!

**Pull request merged**